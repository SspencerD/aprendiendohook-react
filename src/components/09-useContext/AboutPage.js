import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutPage = () => {

     const {user,setUser }= useContext(UserContext)

     const handleClick = () =>{
         setUser({});
     }
    return (
        <div>
             <h1>About</h1>
            <hr/>

            <pre>
                {
                    JSON.stringify(user,null,3)
                }
            </pre>
            <button type="button" 
            className="btn btn-outline-danger"
            onClick={handleClick}
            >
                Logout
            </button>
            
        </div>
    )
}
