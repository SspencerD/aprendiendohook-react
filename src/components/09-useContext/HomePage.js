import React, { useContext } from 'react';
import { UserContext } from './UserContext';


export const Homepage = () => {

    const {user} = useContext(UserContext);
    console.log(user);




    return (
        <div>

            <h1>HomePage</h1>
            <hr />

            <pre>
                {JSON.stringify(user, null ,3)}
            </pre>

        </div>
    )
}
