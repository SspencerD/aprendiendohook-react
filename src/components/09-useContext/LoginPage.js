import React, { useContext } from 'react';
import { MainApp } from './MainApp';
import { UserContext } from './UserContext';


export const LoginPage = () => {


    const {setUser} = useContext(UserContext);


    return (
        <div>
             <h1>Login</h1>
            <hr/>
            <button
            className="btn btn btn-outline-primary"
            onClick={() => setUser( {id:123,
                name:'Santiago',
                lastname:'Spencer',
                email: 'sspencer@recneps.cl'})
             }
            >
                Login

            </button>
            
        </div>
    )
}
