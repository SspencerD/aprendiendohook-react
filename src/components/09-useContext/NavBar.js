import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <NavLink exact activeClassName="active" to="/" className="nav-link" aria-current="page">Home</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink exact activeClassName="active" to="/about" className="nav-link">About</NavLink>
                        </li>
                    </ul>
                    <div className="d-flex">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <NavLink exact activeClassName="active" to="/login" className="nav-link">Login</NavLink>
                        </li>
                        </ul>
                    </div>
                </div>
        </nav>


    )
}
