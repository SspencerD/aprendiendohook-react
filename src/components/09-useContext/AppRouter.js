import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { AboutPage } from './AboutPage';
import { LoginPage } from './LoginPage';
import { Homepage } from './HomePage';
import { NavBar } from './NavBar';


export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />
                <div className="container">
                    <Switch>
                        <Route exact path="/" component={Homepage} />

                        <Route exact path="/about" component={AboutPage} />
                        <Route exact path="/login" component={LoginPage} />

                        <Redirect to="/" />


                    </Switch>

                </div>
            </div>

        </Router>

    )
}
