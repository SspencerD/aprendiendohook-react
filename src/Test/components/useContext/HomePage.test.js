import React from 'react';
import { Homepage } from '../../../components/09-useContext/HomePage';
import { shallow,mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomePage/>', () => {

    
    const user = {
        name:'Santiago',
        lastname:'Spencer',
        email:'sspencer@recneps.cl',

    }
    
    // shallow hace renderizaci√≥n solamente del primer componente,
    // para renderizar el componente completo se utiliza mount
    const wrapper = mount(
    <UserContext.Provider value={{
        user
    }}>
    <Homepage/>
    </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        
    });
    
});