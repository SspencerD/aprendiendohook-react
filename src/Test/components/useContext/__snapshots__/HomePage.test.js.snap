// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomePage/> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomePage
  </h1>
  <hr />
  <pre>
    {
   "name": "Santiago",
   "lastname": "Spencer",
   "email": "sspencer@recneps.cl"
}
  </pre>
</div>
`;
